brewtils/rest/system_client.py

```diff
@@ -203,17 +203,46 @@ def __init__(self, *args, **kwargs):
             )
             kwargs.setdefault("system_name", args[2])
 
-        self._system_name = kwargs.get("system_name")
-        self._version_constraint = kwargs.get("version_constraint", "latest")
-        self._default_instance = kwargs.get("default_instance", "default")
+        system_kwargs = {
+            "system_name": "name",
+            "version_constraint": "version",
+            "default_instance": "instance_name",
+            "system_namespace": "namespace",
+        }
+
+        # brewtils.plugin.CONFIG is already populated
+        # Start by assuming that the system client is targeting self:
+        target_self = True
+
+        # Now we need to figure out if any of the provided kwargs don't match
+        # some sort of loop that checks the value of system kwargs against the CONFIG
+        # If they are different, target_self = False and break
+        for key in system_kwargs.keys():
+            if kwargs.get(key) is None:
+                continue
+            if kwargs.get(key) != brewtils.plugin.CONFIG[system_kwargs[key]]:
+                target_self = False
+                break
```

Comment:
Two things here. First, there's an even better way to do this loop. If you're going to use the key *and* the value when iterating a dict you can use `items()`:

```python
for key, value in system_kwargs.items():
    if kwargs.get(key) is None:
        continue
    if kwargs.get(key) != brewtils.plugin.CONFIG[value]:
        target_self = False
        break
```

Second, it's generally better to avoid `continue` statements it you can. So in this case it's "cleaner" to structure the conditional like this:

```python
for key, value in system_kwargs.items():
    if kwargs.get(key) is not None and kwargs.get(key) != brewtils.plugin.CONFIG[value]:
        target_self = False
        break
```